label_studio_ml/examples/watsonx_llm/model.py

```diff
@@ -0,0 +1,236 @@
+import label_studio_sdk
+import logging
+import os
+import pytesseract
+from ibm_watsonx_ai.foundation_models.utils.enums import ModelTypes
+from ibm_watsonx_ai.metanames import GenTextParamsMetaNames as GenParams
+from label_studio_sdk.label_interface.control_tags import ControlTag, ObjectTag
+from label_studio_sdk.label_interface.object_tags import ImageTag, ParagraphsTag
+from label_studio_sdk.label_interface.objects import PredictionValue
+from ibm_watsonx_ai.foundation_models import ModelInference
+from tqdm import tqdm
+from types import SimpleNamespace
+from typing import List, Dict, Optional, Tuple, Any
+
+from label_studio_ml.model import LabelStudioMLBase
+from label_studio_ml.response import ModelResponse
+
+logger = logging.getLogger(__name__)
+
+
+class WatsonXModel(LabelStudioMLBase):
+    """
+    """
+    WATSONX_CREDENTIALS = {
+        "url": os.getenv('WATSONX_URL', 'https://us-south.ml.cloud.ibm.com'),
+        "apikey": os.getenv('WATSONX_API_KEY')
+    }
+    LABEL_STUDIO_HOST = os.getenv('LABEL_STUDIO_URL', 'http://localhost:8080')
+    LABEL_STUDIO_API_KEY = os.getenv('LABEL_STUDIO_API_KEY')
+    PROJECT_ID = os.getenv('WATSONX_PROJECT_ID')
+    MODEL_TYPE = os.getenv('WATSONX_MODELTYPE')
+    PROMPT_PREFIX = os.getenv('PROMPT_PREFIX', "prompt")
+    USE_INTERNAL_PROMPT_TEMPLATE = bool(int(os.getenv("USE_INTERNAL_PROMPT_TEMPLATE", 1)))
+    DEFAULT_PROMPT = os.getenv('DEFAULT_PROMPT')
+    PROMPT_TEMPLATE = os.getenv('PROMPT_TEMPLATE', '**Source Text**:\n\n"{text}"\n\n**Task Directive**:\n\n"{prompt}"')
+    PROMPT_TAG = "TextArea"
+    SUPPORTED_INPUTS = ("Image", "Text", "HyperText", "Paragraphs")
+
+
+
+    def setup(self):
+        """Configure any parameters of your model here
+        """
+        if self.MODEL_TYPE not in ModelTypes.__members__:
+            raise Exception(f"WATSONX_MODELTYPE must be one of {[model for model in ModelTypes.__members__]}")
+
+        # If you want to specify model parameters, you can do so using this empty parameters dictionary
+        parameters = {
+            GenParams.MAX_NEW_TOKENS: 50,
+        }
+
+        self.model = ModelInference(
```

Comment:
does this code assume reinitialization? how costly is this run? because `setup()` is called quite often, it may slow down predictions if reinitializing every time

Reply:
resolved to not assume reinitialization